src/js/notifications-sw.js

```diff
@@ -0,0 +1,89 @@
+import { default as localforage } from 'localforage';
+
+self.addEventListener('push', event => {
+  function getPayload() {
+    if (event.data) {
+      return Promise.resolve(event.data.json());
+    }
+    return localforage.getItem('deviceId')
+    .then(deviceId => {
+      if (!deviceId) {
+        return null;
+      }
+      return fetch(`./payload/${deviceId}`)
+      .then(response => response.json());
+    });
+  }
+
+  event.waitUntil(
+    getPayload()
+    .then(data => {
+      // TODO add a way to unregister from the back-end
+      // data.command (?)
+      if (data && !data.title) {
+        data = JSON.parse(data);
```

Comment:
I would think so as well, but in Chrome response.json() provides a string - I think I know what it is